### DISCUSSION TITLE
Pi sayısı hesaplama

### DISCUSSION THREAD

**Discussion by sonsuzus** [sonsuzus/BilsemPython#90] [Soru Cevap]
Aşağıdaki formüle göre serinin 10milyonuncu elemanına kadar pi sayısını hesaplayınız.

![b0263215b4d34c850860aa755872523e83a5d848](https://user-images.githubusercontent.com/52765796/225040499-4f79a4f4-1ffd-4d52-b394-9d5517530053.svg)

**Comment by sonsuzus** [comment]
![3d0088f09c917541f81fa50b862917a6383722d4](https://user-images.githubusercontent.com/52765796/225342370-572eb754-4127-4202-8d56-3abe1602a474.svg)
![49757e198774b3dc40d9c72124fa5d3b67896889](https://user-images.githubusercontent.com/52765796/225342520-c77e390d-7388-440e-a7ba-55f43965bc45.svg)
Bunlara göre de hesaplayın.

**Reply by AdaNilPoyraz** [reply]
1. formül ile;
```python
pi = 1
for n in range(1,10000001):
    pi = pi + (((-1)**n)/(2*n+1))
print(4*pi)
```
2. formül ile;
```python
pi = 3
for n in range(1,10000001):
    pi = pi + ((((-1)**n)*-4)/(((2*n+1)**3)-(2*n+1)))
print(pi)
```
3. fomül ile;
```python
pi = 3
for n in range(1,10000001):
    pi = pi + ((((-1)**n)*-4)/((2*n)*(2*n+1)*(2*n+2)))
print(pi)
```

!!NOT!!
Bilgisayarım 10 milyonuncu basamağı hesaplayacak kadar güçlü değil ancak 1000 ile denedim çalıştı.

**Comment by OmerB2** [comment]
```python
t=0
k=1
for i in range(1,10000001,2):
    t=t+k*1/i
    k=-k
t=t*4
print(t)

#diğer formül

t1=0
k1=1
for i1 in range(3,10000001,2):
    t1=t1+k1*4/((i1*i1*i1)-i1)
    k1=-k1
t1=t1+3
print(t1)

#diğer formül

t2=0
k2=1
for i2 in range(2,10000001):
    t2=t2+k2*4/(i2+i2+1+i2+2)
    k2=-k2
t2=t2+3
print(t2)
```

**Comment by AcelyaP** [comment]
İlk formül:
```
pi=0
katsayı=1
for i in range(1,10000001,2):
    pi=pi+katsayı*1/i
    katsayı=-katsayı
pi=pi*4

print(pi)

```
ikinci formül:
```
pi=3
katsayı=1
for i in range(3,10000001,2):
    pi=pi+katsayı*4/(i*i*i-i)
    katsayı=-katsayı

print(pi)
```

Üçüncü formül:
```
pi=3
katsayı=1
for i in range(2,10000001,2):
    pi=pi+katsayı*4/(i*(i+1)*(i+2))
    katsayı=-katsayı

print(pi)
```

**Comment by Musabkaracinar** [comment]
`pisayi = 1

for i in range(1,10000001):
    pisayi = pisayi + (((-1)**n)/(2*n+1))
print(4*pisayi)`

**Comment by Berkek007** [comment]
```
n=0
t=0
pi=0

while n<100:
    
    pi=((1/16)**n)*((4/(8*n+1))-(2/(8*n+4))-(1/(8*n+5))-(1/(8*n+6)))
    t+=pi
    n+=1

print(t)
```
